Stripe/PublicHeaders/STPBackendAPIAdapter.h

```diff
@@ -48,77 +48,20 @@ NS_ASSUME_NONNULL_BEGIN
- succeeds, call `completion(nil)`. Otherwise, call `completion(error)` with the 
- error that occurred.
-
- @param source     The newly-selected default source for the user.
- @param completion call this callback when you're done selecting the new default 
- source for the customer on your backend. For example, `completion(nil)` (if 
- your call succeeds) or `completion(error)` if an error is returned.
- */
-- (void)selectDefaultCustomerSource:(id<STPSourceProtocol>)source completion:(STPErrorBlock)completion;
-
-@optional
-
-/**
- Deletes the given source from the customer.
+ Retrieves a list of Payment Methods attached to a customer.
  
  If you are implementing your own <STPBackendAPIAdapter>:
- On your backend, retrieve the Stripe customer associated with your logged-in user.
- Then, call the Delete Card method ( https://stripe.com/docs/api#delete_card )
- specifying id to be the value of source.stripeID. If this API call
- succeeds, call `completion(nil)`. Otherwise, call `completion(error)` with the
- error that occurred.
-
- @param source    The source to delete from the customer
- @param completion call this callback when you're done deleting the source from
- the customer on your backend. For example, `completion(nil)` (if your call
- succeeds) or `completion(error)` if an error is returned.
- */
-- (void)detachSourceFromCustomer:(id<STPSourceProtocol>)source completion:(nullable STPErrorBlock)completion;
-
-/**
- Sets the given shipping address on the customer.
+ Call the list method ( https://stripe.com/docs/api/payment_methods/lists )
+ with the Stripe customer. If this API call succeeds, call `completion(paymentMethods)`
+ with the list of PaymentMethods. Otherwise, call `completion(error)` with the error
+ that occurred.
  
- If you are implementing your own <STPBackendAPIAdapter>:
- On your backend, retrieve the Stripe customer associated with your logged-in user.
- Then, call the Customer Update method ( https://stripe.com/docs/api#update_customer )
- specifying shipping to be the given shipping address. If this API call succeeds, 
- call `completion(nil)`. Otherwise, call `completion(error)` with the error that occurred.
-
- @param shipping   The shipping address to set on the customer
- @param completion call this callback when you're done updating the customer on
- your backend. For example, `completion(nil)` (if your call succeeds) or
+ @param completion  Call this callback with the list of Payment Methods attached to the
+ customer.  For example, `completion(paymentMethods)` (if your call succeeds) or
  `completion(error)` if an error is returned.
-
- @see https://stripe.com/docs/api#update_customer
+ 
  */
-- (void)updateCustomerWithShippingAddress:(STPAddress *)shipping completion:(nullable STPErrorBlock)completion;
-
-#pragma mark - Payment Method
+- (void)listPaymentMethodsForCustomerWithCompletion:(nullable STPPaymentMethodsCompletionBlock)completion;
```

Comment:
I think this completion block should be nonnull, because it doesn't do anything otherwise right?

Reply:
`STPCustomerContext` passes nil to this and `retrieveCustomer` (also has a nullable completion block) because its implementation caches the response, and it calls them for that side effect in the init.  Conceivably someone could be using these in the same way..

Reply:
bleh -- kinda hacky to call like this for the side-effect, but if it's a potential breaking change we can keep as is for now